csaf_2.0/guidance/sbom-matching-interface.md

```diff
@@ -0,0 +1,91 @@
+| `Matcher`            | The main interface for matching. It provides methods to match a list of CSAF document against a database of SBOMs. It contains the following properties: <br/>- `documents`: The CSAF documents to match against. <br/>The following operations MUST be supported:<br/>- `matchComponent(component, threshold)`: Matches the `documents` against one component in the SBOM and returns a `Match`<br/>- `match(sbom, threshold)`: Matches the `documents` against the complete SBOM and returns a `Match`<br/>- `matchDatabase(database, threshold)`: Matches the `documents` against the complete SBOM `database` and returns a `Match` |
+| `Match`              | The result of a matching operation. It contains the matched SBOMs and their corresponding CSAF document. It contains the following properties:<br/>- `document`: The CSAF document this match was generated from<br/>- `product`: The CSAF product that generated the match<br/>- `matchedComponent`: The SBOM component that is matched to this document<br/>- `confidence`: The confidence of this match                                                                                                                                                                                                                              |
+| `MatchingConfidence` | Since matching ist mostly not a 100 % match, we want to give some indication of the confidence we have in the matching (ranging from `0.0` to `1.0`)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
+
+## Confidence Levels
+
+Each value between `0.0` and `1.0` can be used to indicate a confidence level. However, it is recommended to use the values defined in the table below or a multiplication of them. For example, if a `CaseInsensitiveMatch` is used in combination with a `DifferentSources`, the resulting confidence level would be `0.95 * 0.90 = 0.855`. The following table defines the known confidence levels and their meaning:
+
+| Short Identifier                 | Confidence Level | Comment                                                                                                                                                                                                                                                             |
```

Comment:
Maybe here a definition list would be better adapted? The primary key is the confidence level, and the short identifier is a memory helper / alias, no?

Like:

```
0.00 (`DefinitelyNoMatch`)
:    Indicates that the component is definitely not matched to the document.

0.50 (PartialStringMatch`)
:    Indicates that a string property (e.g., the vendor name) of the product partially matches the component's string property. 

...
```